### PR TITLE
Identify the current primary replica and configure ag on it

### PR DESCRIPTION
**Pull request by spetrosi**
Previously, the role configured AG on the server that has the
`mssql_ha_replica_type: primary` variable set.
However, i the case of fail over the primary replica moves to a
different server.
With this change, the role identifies the current primary server and
configures AG on it.

### CODE REVIEW

**Review by richm — APPROVED** [review]

**Review by richm — APPROVED** [review]

File: tasks/main.yml [hunk 1]
```diff
@@ -668,36 +668,32 @@
...
+        - name: Grant permissions to the {{ mssql_ha_login }} login
+          vars:
+            __mssql_input_sql_file: grant_permissions_to_ha_login.j2
+          include_tasks: input_sql_file.yml
+
+        - name: Back up and replicate databases
+          vars:
+            __mssql_input_sql_file: replicate_db.j2
+          include_tasks: input_sql_file.yml
+          when: mssql_ha_db_names is not none
```
Comment (richm) [inline tasks/main.yml:696]: I guess this will change when https://github.com/linux-system-roles/mssql/pull/116 is merged

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest pending]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
Fedora 36 tests fails with errors similar to what I saw on RHEL 9 when downloading mssql-server from RHEL 8 repositories (Microsoft does not provide repos for RHEL 9 now). It is mssql-server issue, it does not support Fedora officially so there is not much I can do.

**Pull request merged**